### PR TITLE
feat: add --host to CLI

### PR DESCRIPTION
**Pull request by mscolnick**

### CODE REVIEW

File: marimo/_server/api/lifespans.py [hunk 1]
```diff
@@ -102,8 +102,9 @@ async def lsp(app: Starlette) -> AsyncIterator[None]:
 
 @contextlib.asynccontextmanager
 async def open_browser(app: Starlette) -> AsyncIterator[None]:
-    session_mgr = get_manager()
-    url = f"http://localhost:{session_mgr.port}"
+    host = app.state.host
+    port = app.state.post
```
Comment (akshayka) [inline marimo/_server/api/lifespans.py:106]:
```suggestion
    port = app.state.port
```

File: marimo/_cli/cli.py [hunk 2]
```diff
@@ -318,6 +339,13 @@ def recover(name: str) -> None:
     type=int,
     help="Port to attach to.",
 )
+@click.option(
+    "--host",
+    default="127.0.0.1",
```
Comment (akshayka) [inline marimo/_cli/cli.py:344]: @mscolnick Can we continue to use `localhost` as the default? Or at least print `localhost` at the CLI when `host==127.0.0.1`? I think that's friendlier for our users.

**Review by akshayka — COMMENTED** [review]
Left 1 comment. CI was failing, maybe due to a typo, which I fixed

File: marimo/_cli/cli.py (hunk 2 above)
Comment (mscolnick) [reply marimo/_cli/cli.py:344]: Most modern web frameworks have moved from `localhost` to explicitly `127.0.0.1`. I've seen this on both Python (starlette, fastapi) and Node (vite). I am sure there is good reason, so I would advise to not make localhost the default.

For printing, most every case localhost will be aliased to `127.0.0.1`, but it is possible that the user had changed the alias. I'm fine to still print `localhost` as the default if you think it is friendlier

**Review by akshayka — APPROVED** [review]

**Pull request merged**